runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -172,7 +188,7 @@ private void appendCheckpointOnSnapshot(Token snapshotTimestamp) {
             log.info("appendCheckpoint: Started checkpoint for {} at snapshot {}", streamId, snapshotTimestamp);
             ICorfuSMR<T> corfuObject = (ICorfuSMR<T>) this.map;
             Set<Map.Entry> entries = this.map.entrySet();
-            long vloVersion = corfuObject.getCorfuSMRProxy().getVersion();
+            long vloVersion = snapshotTimestamp.getSequence();
```

Comment:
This needs a comment about why its reasonable to "change" the vlo version

Reply:
Done.

Reply:
Instead of reassigning the vloVersion variable, you can check if long vloVersion == corfuObject.getCorfuSMRProxy().getVersion(); and maybe log something ? do we even care?